### ISSUE TITLE
bson.Objectid issue in node 12.*+

### ISSUE THREAD

**Issue by liars2** [mongodb/js-bson#324]
I am using node 12.* verison, and i found issue by using os modules

in line 3. just calling hostname and create random 3-byte value.

```
'use strict';

const hostname = require('os').hostname;
const fnv1a24 = require('./fnv1a').fnv1a24;

/**
 * Machine id.
 *
 * Create a random 3-byte value (i.e. unique for this
 * process). Other drivers use a md5 of the machine id here, but
 * that would mean an asyc call to gethostname, so we don't bother.
 * @ignore
 */
const MACHINE_ID = fnv1a24(hostname);
```

but it's wrong.

[please look at this docs.](https://nodejs.org/dist/latest-v8.x/docs/api/os.html#os_os_hostname)

The `os.hostname` is not default string, it's a function.

**Comment by liars2** [comment]
sorry my bad

**Comment by SayakMukhopadhyay** [comment]
@liars2 Could you elaborate if you solved this issue. I am facing a similar situation and haven't got any clue to solve this.

**Comment by SayakMukhopadhyay** [comment]
Welp, this issue has been fixed in v4. For anyone strolling in here, check your dependencies, you need v4+ for this issue to be resolved. If you don't have a dependency on bson, check your error logs to see which modules in your `node_modules` do have that as a dependency and if that dependecy is updated to v4+ or not.

For me, I had a module which was still at v3.0.2. Upgrading the dependency made everything work just fine.

**Comment by lionep** [comment]
Can't a v3.0.3 be released with this fix ?  
The v4 has other breaking changes.

**Comment by ggondim** [comment]
Somehow the `3.0` branch fixed this issue for 3.x versions, but the available published tags (3.0.0, 3.0.1 and 3.0.2) doesn't have it: https://github.com/mongodb/js-bson/blob/3.0/lib/bson/objectid.js.

I'm forking mongodb-extjson project to fix it, but someone needs to make a `3.0.4` with this fix here.

**Comment by ggondim** [comment]
I created a pull request for it https://github.com/mongodb/js-bson/pull/368
